qubes-dist-upgrade.sh

```diff
@@ -212,6 +213,20 @@ max_concurrency="${max_concurrency:-4}"
 # Run prechecks first
 update_prechecks
 
+# Automatically add qubes providing dom0 input devices to remain powered on
+if which xinput > /dev/null; then
+    read -r -a xinput_qubes <<< "$(xinput list --name-only | sed -n 's/\(^[a-zA-Z][a-zA-Z0-9_-]*\):.*$/\1/p')"
+    for qube in "${xinput_qubes[@]}"
+    do
+        if qvm-check -q "$qube"; then
+            echo "INFO: Found $qube providing input devices. It will be kept running."
+            extra_keep_running+=( "$qube" )
```

Comment:
This (currently) won't be enough to keep it running in stage 3. See the first task in the ticket description.

And also, after stage 3 no new qrexec service will work, which means if somebody unplug USB mouse/keyboard, plugging it back wont work. Maybe include it in the warning too?